controllers/operatorscaler_controller.go

```diff
@@ -0,0 +1,329 @@
+					client.InNamespace(r.OperatorNamespace),
+					client.MatchingLabels(map[string]string{key: ""}),
+				)
+				if err != nil {
+					r.log.Error(err, "failed listing csvs with label", "label", key)
+					multierr.AppendInto(&returnErr, err)
+				} else {
+					for i := range csvList.Items {
+						if err = r.updateCsvDeplymentsReplicas(&csvList.Items[i]); err != nil {
+							r.log.Error(err, "failed updating csvs replica")
+							multierr.AppendInto(&returnErr, err)
+						}
+					}
+				}
+			}
+		}
+	}
+
+	r.log.Info("successfully completed reconcileOperators")
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) updateCsvDeplymentsReplicas(csv *opv1a1.ClusterServiceVersion) error {
+
+	var updateRequired bool
+	for i := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		deploymentSpec := &csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec
+		if *deploymentSpec.Replicas < 1 {
+			deploymentSpec.Replicas = ptr.To(int32(1))
+			updateRequired = true
+		}
+	}
+
+	if updateRequired {
+		if err := r.Client.Update(r.ctx, csv); err != nil {
+			r.log.Error(err, "failed updating csv replica", "csvName", csv.Name)
+			return err
+		}
+		r.log.Info("csv updated successfully", "csvName", csv.Name)
+	}
+
+	return nil
+}
+
+func (r *OperatorScalerReconciler) reconcileDynamicWatchers() error {
+	r.log.Info("entering reconcileDynamicWatchers")
+
+	for i := range ResourceMappingList {
+		resourceMapping := &ResourceMappingList[i]
+
+		if !r.kindsBeingWatched[resourceMapping.Kind] {
+
+			crd := &extv1.CustomResourceDefinition{}
+			crd.Name = resourceMapping.CrdName
+			if err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(crd), crd); client.IgnoreNotFound(err) != nil {
+				r.log.Error(err, "failed getting crd", "crdName", resourceMapping.CrdName)
+				return err
+			} else if err == nil {
+				r.log.Info("adding dynamic watch", "kind", resourceMapping.Kind)
+
+				err := r.controller.Watch(
+					source.Kind(
+						r.cache,
+						client.Object(
+							&metav1.PartialObjectMetadata{
+								TypeMeta: metav1.TypeMeta{
+									APIVersion: resourceMapping.ApiVersion,
+									Kind:       resourceMapping.Kind,
+								},
+							},
+						),
+						&handler.EnqueueRequestForObject{},
+						// Trigger the reconcile for creation events of the object.
+						// This ensures the replicas in the CSV are scaled up based on the presence of Custom Resource (CR).
+						createOnlyPredicate,
+					),
+				)
+				if err != nil {
+					r.log.Error(err, "failed adding dynamic watch", "kind", resourceMapping.Kind)
+					return err
+				}
+
+				r.kindsBeingWatched[resourceMapping.Kind] = true
+			}
+		}
+	}
+
+	r.log.Info("successfully completed reconcileDynamicWatchers")
+	return nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *OperatorScalerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	// It will keep track or the crds that we care about
+	crdMap := make(map[string]bool)
```

Comment:
There is no need for a make here, the type is hard coded
```suggestion
	crdMap := map[string]bool{}
```

Reply:
Done